session106/XMap/XMap/Model/TFLFetcher.swift

```diff
@@ -59,6 +59,14 @@ class TFLFetcher: NSObject {
             // Received Data for parsing
             // Do some work...Mahyad?
             
+            do {
+                let decoder = JSONDecoder()
+                let arrivalTube = try decoder.decode(ArrivalTube.self, from: responseJSON as! Data)
+                print(arrivalTube)
+            } catch {
+                print("fuck me")
+            }
```

Comment:
So, here, because the responseJSON is Any, with based on the JSON response scheme I sent you is an array of dictionaries. So we have to cast it into an array first. Then iterate through it and convert each dictionary into Data. You can force it to Data! Use JSONSerialization. Finally you can add the parsed Classes into an array and return it in the handler.